client/components/phone-input/test/phone-number__misc.test.ts

```diff
@@ -0,0 +1,131 @@
+import { groupBy, pickBy } from 'lodash';
+import { countries } from '../data';
+import {
+	indexOfLongestCommonSuffix,
+	indexOfStrictSubsequenceEnd,
+	nonDigitsAtStart,
+	numDigitsBeforeIndex,
+	getUpdatedCursorPosition,
+} from '../phone-number';
+
+describe( 'Data assertion', function () {
+	// Generates a list of countries that share the dial code, with following properties:
+	// isoCode, dialCode, nationalPrefix, patternRegion, priority
+	const countriesShareDialCode = pickBy(
+		groupBy( Object.values( countries ), 'dialCode' ),
+		( val ) => val.length > 1
+	);
+
+	describe( 'countries sharing dial code should have priority data', function () {
+		Object.entries( countriesShareDialCode ).map( ( [ dialCode, countriesWithDialCode ] ) => {
+			describe( 'Dialcode: ' + dialCode, function () {
+				countriesWithDialCode.forEach( function ( country ) {
+					test( `${ country.isoCode } has priority`, function () {
+						expect( country.priority ).toBeTruthy();
+					} );
+				} );
+			} );
+		} );
+	} );
+} );
+
+describe( 'indexOfLongestCommonSuffix', function () {
+	test.each( [
+		{ array1: [], array2: [], expected: [ 0, 0 ] },
+		{ array1: [ 'a' ], array2: [], expected: [ 1, 0 ] },
+		{ array1: [], array2: [ 'a' ], expected: [ 0, 1 ] },
+		{ array1: [ 'a' ], array2: [ 'a' ], expected: [ 0, 0 ] },
+		{ array1: [ 'a', 'b', 'c' ], array2: [ 'a', 'b', 'c' ], expected: [ 0, 0 ] },
+		{ array1: [ 'a', 'b', 'c' ], array2: [ 'a', 'b', 'c', 'd' ], expected: [ 3, 4 ] },
+		{ array1: [ 'a', 'b', 'c' ], array2: [ 'b', 'c' ], expected: [ 1, 0 ] },
+		{ array1: [ 'x', 'y', 'z' ], array2: [ 'a', 'b', 'c', 'd' ], expected: [ 3, 4 ] },
+	] )(
+		`Correct index for longest suffix is found for inputs $array1, $array2`,
+		function ( { array1, array2, expected } ) {
+			const result = indexOfLongestCommonSuffix( array1, array2 );
+			expect( result ).toStrictEqual( expected );
+		}
+	);
+} );
+
+describe( 'indexOfStrictSubsequenceEnd', function () {
+	test.each( [
+		{ array1: [ 'a', 'b' ], array2: [ 'a', 'b', 'c' ], expected: 2 },
+		{ array1: [ 'a', 'b' ], array2: [ 'a', 'd', 'b', 'c' ], expected: 3 },
+	] )(
+		`Correct index for subsequence is found for inputs $array1, array2`,
```

Comment:
I think you're missing a "$" here! `$array2`

Reply:
Fixed in e10751336012e40c886f517b08cc4f2cbe15d2e0.